q2_metadata/normalization/_prepare_default_rules.py

```diff
@@ -0,0 +1,343 @@
+    return real_name, add_on_info
+
+
+def get_meta_from_template(
+        agp_suppl_pd: pd.DataFrame,
+        md_temp_pd: pd.DataFrame) -> (dict, dict):
+    """
+    :param agp_suppl_pd: AGP metadata descriptions table (per variable name)
+    :param md_temp_pd: metadata template table
+    :return:
+        files_data:
+            Keys    = variable
+            Values  = possible values examples/defaults
+                e.g. {'Expected values': ['ex: 1001'],
+                      'Missing values': ['Not provided'],
+                      'Blank value': ['BLANK[identifier]']}
+        variables_renamed:
+            Keys    = variable
+            Values  =
+    """
+    real_name, add_on_info = get_template_vs_agp_inconsistencies()
+    files_data = {}
+    variables_renamed = {}
+    # for each enumerated template metadata's variable (in "column" column) and its table data
+    for variable_, md_temp in md_temp_pd.groupby('column'):
+        # remove the metadata variable names column, transpose, and rename
+        md_temp_t = md_temp.drop(columns='column').T.reset_index()
+        md_temp_t.columns = ['col', variable_]
+        md_temp_t = md_temp_t.loc[(md_temp_t[variable_] != '(pick a) biome') &
+                                  (~md_temp_t[variable_].isna())]
+        # collect the unique possible values from template
+        md_temp_d = get_md_temp_dict(md_temp_t)
+        # edit key in template metadata to the matching name in qiita
+        variable = real_name.get(variable_, variable_)
+        # collect the possible values for the current variable
+        files_data[variable] = md_temp_d
+        # apply changes to roll back to AGP standards
+        variables_renamed = compare_with_agp(
+            variable,
+            agp_suppl_pd,
+            variables_renamed
+        )
+    return files_data, variables_renamed
+
+
+def get_factors(cell: str) -> list:
+    """
+    Split a metadata factor and remove the trailing quotes
+    :param cell: metadata factor (i.e. table cell content)
+    :return: content split on the pipe character
+    """
+    cell_split = [x.strip().strip('"').strip("'") for x in cell.split('|')]
+    return cell_split
+
+
+def update_derivative_source(
+        cur_agp_suppl_pd: pd.DataFrame,
+        variable: str,
+        derivatives: dict,
+        sources: dict) -> (dict, dict):
+    """
+    Add the current variable's values for AGP definitions of:
+    - Derivative columns
+    - Source columns
+    :param cur_agp_suppl_pd: AGP supplementary info reduce to the current variable.
+    :param variable: current variable name.
+    :param derivatives: to be update here.
+    :param sources: to be update here.
+    :return: updated derivatives and sources dicts.
+    """
+    cur_agp_suppl_pd.set_index('col', inplace = True)
+    cur_derivative = str(cur_agp_suppl_pd.loc['Derivative columns', variable])
+    if cur_derivative != 'nan':
```

Comment:
Are the nan values read in as floats or strings? Since a float nan != another float nan, just want to make sure this condition gets met properly